pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java

```diff
@@ -346,9 +352,38 @@ public void start()
       timeSeriesRequestHandler = new TimeSeriesRequestHandler(_brokerConf, brokerId, _routingManager,
           _accessControlFactory, _queryQuotaManager, tableCache, queryDispatcher);
     }
+
+    LOGGER.info("Initializing PinotFSFactory");
+    PinotFSFactory.init(_brokerConf.subset(CommonConstants.Broker.PREFIX_OF_CONFIG_OF_PINOT_FS_FACTORY));
+
+    LOGGER.info("Initialize ResultStore");
+    PinotConfiguration resultStoreConfiguration =
+        _brokerConf.subset(CommonConstants.CursorConfigs.PREFIX_OF_CONFIG_OF_RESULT_STORE);
+    try {
+      ResponseSerde responseSerde = ResponseSerdeService.getInstance().getResponseSerde(
+          resultStoreConfiguration.getProperty(CommonConstants.CursorConfigs.RESULT_STORE_SERDE,
+              CommonConstants.CursorConfigs.DEFAULT_RESULT_SERDE));
+      responseSerde.init(resultStoreConfiguration.subset(CommonConstants.CursorConfigs.RESULT_STORE_SERDE)
+          .subset(responseSerde.getType()));
+
+      String expirationTime = getConfig().getProperty(CommonConstants.CursorConfigs.RESULTS_EXPIRATION_INTERVAL,
+          CommonConstants.CursorConfigs.DEFAULT_RESULTS_EXPIRATION_INTERVAL);
+
+      _resultStore = (AbstractResponseStore) ResponseStoreService.getInstance().getResultStore(
```

Comment:
This doesn't look pretty. Here we are effectibly enforcing that `ResponseStoreService.getInstance().getResultStore(whatever)` should always return an `AbstractResponseStore`. If that is the case we should change the `getResultStore` method to enforce that at compile time. Alternatively we could relax the requirements in this class to be able to work with a `ResponseStore` (the interface)

Reply:
This is a consequence of not being able to import `BrokerResponse` in the SPI package. `ResponseStoreService` is in the SPI package. Some methods that should have been in `ResponseStore` are in `AbstractResponseStore` because the methods need `BrokerResponse`. 

The only option is to not define the interfaces in the SPI package and move them to commons package.

Reply:
That is even worse, right?

IIRCU your response, it means that the `ResponseStore` defined in SPI is not useful, given implementations will need to extend `AbstractResponseStore`, which is not in SPI. That is something we need to improve, otherwise plugins that provide their own `ResponseStore` will need to depend on `pinot-common`, which is not what we want. But even if we accept that, as you said, we could remove `ResponseStore`.

Reply:
Removing `ResponseStore` will imply moving `ResponseStoreService` to `pinot-common` as well. Also `ResponseStore` and `ResponseSerde` are a pair. They should not be separated. 

`ResponseStore` plugins have to depend on `pinot-common`. I have tried to move `BrokerResponse`, `ResultTable` and `DataSchema` to SPI and it is too disruptive.

Reply:
We didn't find a better way to abstract the code, so we are going to go with this cast and add some extra comments in `ResposeStore`.